### DISCUSSION TITLE
Where can I change the thread timeout setting?

### DISCUSSION THREAD

**Discussion by dannyshin76** [Budibase/budibase#11361] [Help] [answered]
I'm using Budibase self-host app (v2.7.36) with REST API as my data sources.
Sometimes my API takes more than 10 seconds to respond, and then the data source fails with this error:
![image](https://github.com/Budibase/budibase/assets/93232582/4563b058-6a39-4216-ae15-c42127f03957)

I remember I didn't get this timeout before upgrading to v2.7.36. Is there a setting that I can change on my hosting to make the timeout longer?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @dannyshin76,

The answer here should help you; https://github.com/Budibase/budibase/discussions/11360#discussioncomment-6549327

I hope this helps.

**Reply by dannyshin76** [reply]
Thanks! Will try that.

**Comment by DeveloperMDCM** [comment]
the solution
https://github.com/DeveloperMDCM/budibase-timeout-fix
